typhon/math/array.py

```diff
@@ -157,26 +155,42 @@ def mad_outliers(arr, cutoff=10, mad0="raise"):
     """
 
     if arr.ptp() == 0:
-        return numpy.zeros(shape=arr.shape, dtype="?")
+        return np.zeros(shape=arr.shape, dtype="?")
 
-    ad = abs(arr - numpy.ma.median(arr))
-    mad = numpy.ma.median(ad)
+    ad = abs(arr - np.ma.median(arr))
+    mad = np.ma.median(ad)
     if mad == 0:
         if mad0 == "raise":
             raise ValueError("Cannot filter outliers, MAD=0")
         elif mad0 == "perc":
             # try other percentiles
-            perc = numpy.r_[
-                numpy.arange(50, 99, 1),
-                numpy.linspace(99, 100, 100)]
+            perc = np.r_[np.arange(50, 99, 1), np.linspace(99, 100, 100)]
             pad = scipy.stats.scoreatpercentile(ad, perc)
-            if (pad==0).all(): # all constant…?
+            if (pad == 0).all():  # all constant…?
                 raise ValueError("These data are weird!")
             p_i = pad.nonzero()[0][0]
             cutoff *= (100 - 50) / (100 - perc[p_i])
             return (ad / pad[p_i]) > cutoff
-    elif mad is numpy.ma.masked:
+    elif mad is np.ma.masked:
         # all are masked already…
-        return numpy.ones(shape=ad.shape, dtype="?")
+        return np.ones(shape=ad.shape, dtype="?")
     else:
         return (ad / mad) > cutoff
+
+
+def argclosest(array, value, retvalue=False):
+    """Returns the index of the closest value in array.
+
+    Parameters:
+        array (ndarray): Input array.
+        value (float): Value to compare to.
+        retvalue (bool): If True, return the index and the closest value.
+
+    Returns:
+        int, float:
+        Index of closest value, Closest value (if ``retvalue`` is True)
+
+    """
+    idx = np.abs(array - value).argmin()
```

Comment:
I'd suggest to use asarray to make this compatible with lists at basically no extra cost:

```python
idx = np.abs(np.asarray(array) - value).argmin()
```

Also one thing to keep in mind is that this algorithm does prioritize the lower index if the distance to each neigbor is the same. This leads to mathematically unexpected behaviour for .5 numbers:

```python
value=1.5
typhon.math.array.argclosest([1.,2.,3.], value, retvalue=True)[1]
# Returns: 1.0
round(value)
# Returns: 2

Reply:
I totally agree with the ``asarray`` fix. I do not see a problem with the "distance" problem, should I simply add a note to the docstring, so that users are aware?

Reply:
Nah, it's fine. That's why I said "keep it in mind", not "put it in the docs". ;-) Ok to merge.